src/engine/Globals.ml

```diff
@@ -47,10 +58,9 @@ let reset () =
   Core_profiling.mode := Core_profiling.MNo_info;
   AST_generic_equals.busy_with_equal := AST_generic_equals.Not_busy;
   Rule.last_matched_rule := None;
+  reset_pro_hooks ();
   (* TODO?
-   * - semgrep-pro hooks?
-   * - Match_patterns.last_matched_rule?
```

Comment:
do you know about Match_patterns.last_matched_rule? Because we have also Rule.last_matched_rule
but I don't know the difference (I didn't investigate).

Reply:
I saw your comment while investigating another ticket and I myself don't know how it is being used. 😕 I would need to investigate too.